pkg/datapath/iptables/iptables.go

```diff
@@ -72,13 +99,13 @@ func getFeedRule(name, args string) []string {
 	return append(argsList, ruleTail...)
 }
 
-func (c *customChain) add() error {
+func (c *customChain) add(waitArgs []string) error {
 	var err error
 	if option.Config.EnableIPv4 {
-		err = runProg("iptables", []string{"-t", c.table, "-N", c.name}, false)
+		err = runProg("iptables", append(waitArgs, "-t", c.table, "-N", c.name), false)
```

Comment:
Instead of passing `waitArgs` to each occurrence of `runProg`, we could refactor each function which calls it in a way that a function would return list of rules, and then a higher-level function would invoke `iptables` with `waitArgs`.